plugins/omar-wms-plugin/src/test/groovy/omar/wms/WmsControllerSpec.groovy

```diff
@@ -0,0 +1,87 @@
+package omar.wms
+
+import grails.testing.services.ServiceUnitTest
+import spock.lang.*
+import omar.wms.WmsController
+
+import java.nio.Buffer
+
+class WmsControllerSpec extends Specification
+{
+
+    /*
+        Here we want to test WebMappingService.getMap() method, mocking the calls to external services
+     */
+    void "test getMap(getPsm=true)"()
+    {
+        def wms = Spy(WebMappingService)
+
+        GetMapRequest getMapRequest = new GetMapRequest() // required param for WebMappingService.getMap()
+        getMapRequest.width = 256
+        getMapRequest.height = 256
+        getMapRequest.bbox = "1.1,1.1,1.1,1.1"
+        getMapRequest.service = "WMS"
+        getMapRequest.version = "1.1.1"
+        getMapRequest.request = "getPsm"
+        getMapRequest.srs = "EPSG:4326"
+        getMapRequest.crs = null
+        getMapRequest.bbox = '147.3211669921875,-42.879638671875,147.32391357421875,-42.87689208984375'
+        getMapRequest.format = 'image/png'
+        getMapRequest.layers = 'omar:raster_entry'
+        getMapRequest.styles = "{\"bands\":\"default\",\"histCenterTile\":false,\"histOp\":\"auto-minmax\",\"resamplerFilter\":\"bilinear\",\"nullPixelFlip\":false}"
+        getMapRequest.transparent = true
+        getMapRequest.filter = 'INTERSECTS(ground_geom,POLYGON((146.9285294926882 -42.96513088145454, 146.9285294926882 -42.74156864994309, 147.4379233272598 -42.74156864994309, 147.4379233272598 -42.96513088145454, 146.9285294926882 -42.96513088145454)))'
```

Comment:
I suggest pulling a lot of these params and filters up as members of the class so they can be re-used across a number of tests.

Reply:
Done